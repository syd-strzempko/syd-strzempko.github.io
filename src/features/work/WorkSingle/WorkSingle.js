import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Layout from '../../../components/Layout';
import { selectSingle } from '../workSlice';
import styles from './WorkSingle.module.css';

export const WorkSingle = () => {
  let { id } = useParams();
  const single = useSelector((state) => selectSingle(state, id));
  return (
    <Layout header={single.title ? single.org : single.project} returnURL={'/work'}>
      <div className={styles.title}>{single.title ? single.title : single.org}</div>
      <div className={styles.subTitle}>{single.span}</div>
      {single.stack && <div>Stack: {single.stack}</div>}
      <div className={styles.listWrapper}>
        {single.details.map((detail, i) => {
          return (
            <div key={i} className={styles.listItem}>
              <div className={styles.bullet} />
              <span>{detail}</span>
            </div>
          );
        })}
      </div>
      {single.link && <Link to={single.link} className={styles.linkButton}>link <div>Â»</div></Link>}
    </Layout>
  );
}