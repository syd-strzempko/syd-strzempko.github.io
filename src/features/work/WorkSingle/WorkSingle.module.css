.title {
    text-align: center;
    font-size: 1.25rem;
    padding-bottom: 10px;
}
.subTitle {
    color: var(--beige1);
    font-style: italic;
    padding-bottom: 5px;
}
.listItem{
    display: flex;
    align-items: center;
    gap: 10px;
    padding: 5px 0;
}
.bullet {
    background-color: var(--link-active);
    min-width: 5px;
    height: 5px;
    margin: 5px;
}
.linkButton {
    align-self: self-end;
    font-variant: small-caps;
    display: inline-flex;
    div {
        padding: 4px 0 0 3px;
        transform: rotate(-90deg);
    }
}