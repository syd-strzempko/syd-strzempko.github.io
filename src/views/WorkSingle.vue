 <template>
  <div class="content">
    <router-link to="/work">
      <span class="iconlink">â†©</span>
    </router-link>
    <div v-if="data">
      <fragment v-if="data.title">
        <h1>{{ data.title }}</h1>
        <a v-if="data.link" :href="data.link">
          <h2>{{ data.org }}</h2>
        </a>
        <h1 v-else>{{ data.org }}</h1>
        <h2>{{ data.project }}</h2>
      </fragment>
      <fragment v-else>
        <a v-if="data.link" :href="data.link">
          <h1>{{ data.project }}</h1>
        </a>
        <h1 v-else>{{ data.project }}</h1>
        <h2>{{ data.org }}</h2>
      </fragment>
      <h2>{{ data.span }}</h2>
      <ul>
        <li
          v-for="(detail, index) in data.details" v-bind:key="index">
          {{detail}}
        </li>
      </ul>
    </div>
    <h1 v-else>no job here :(</h1>
  </div>
</template>

<script>
import { Fragment } from 'vue-fragment'

export default {
  name: 'WorkSingle',
  props: {
    data: Object
  },
  components: {
    Fragment,
  }
}
</script>
