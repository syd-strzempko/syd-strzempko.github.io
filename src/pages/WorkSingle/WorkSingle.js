import { Fragment } from 'react';
import Layout from '../../components/Layout';
import NavBar from '../../components/NavBar';
import './WorkSingle.css';

const WorkSingle = () => {
  return (
    <Layout>
      <NavBar />
      <Fragment>WorkSingle</Fragment>
    </Layout>
  );
}

export default WorkSingle;