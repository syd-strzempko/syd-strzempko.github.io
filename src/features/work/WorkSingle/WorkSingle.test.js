import { WorkSingle } from './WorkSingle';
import { MemoryRouter, useParams } from 'react-router-dom';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
// import { render, screen } from '@testing-library/react';

// Mock the useParams hook
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

// Mock useSelector hooks - redux-mock-store
jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
}));

jest.mock('../../../components/Circles/Circles');

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router
describe('WorkSingle', () => {
    beforeEach(() => {
        useSelectorMock.mockImplementation(selector => selector(mockStore));
    })
    afterEach(() => {
        useSelectorMock.mockClear();
    })

    const useSelectorMock = reactRedux.useSelector;
    const mockStore = {
        work: {
            jobs: [
                {
                    url: 'job_test',
                    org: 'test',
                    project: 'test',
                    title: 'test',
                    link: '/test',
                    span: 'test',
                    details: ['lorem ipsum etc', 'lorem ipsum esque']
                }
            ],
            projects: [
                {
                    url: 'project_test',
                    org: 'test',
                    project: 'test',
                    title: 'test',
                    link: '/test',
                    span: 'test',
                    details: ['lorem ipsum etc', 'lorem ipsum esque']
                }
            ],
            images: [
                {  
                    pathLong: 'mock_long.jpg',
                    pathShort: '/job_test/mock_short.jpg',
                    id: 0,
                    compressed: {
                        pathLong: 'mock_long.jpg',
                        pathShort: 'mock_short.jpg'
                    }
                },
                {  
                    pathLong: 'mock_long.jpg',
                    pathShort: '/project_test/mock_short.jpg',
                    id: 0,
                    compressed: {
                        pathLong: 'mock_long.jpg',
                        pathShort: 'mock_short.jpg'
                    }
                }
            ],
        }
    };

    test('renders correctly with job', () => {
        // Set the desired dynamic value for testing
        // useParams.mockReturnValue({ id: 'job_test' });
        jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ id: 'job_test' });

        let view = renderer.create(
            <MemoryRouter>
                <WorkSingle />
            </MemoryRouter>
        );
        expect(view.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with project', () => {
        // Set the desired dynamic value for testing
        // useParams.mockReturnValue({ id: 'project_test' });
        jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ id: 'project_test' });

        let view = renderer.create(
            <MemoryRouter>
                <WorkSingle />
            </MemoryRouter>
        );
        expect(view.toJSON()).toMatchSnapshot();
    });
});