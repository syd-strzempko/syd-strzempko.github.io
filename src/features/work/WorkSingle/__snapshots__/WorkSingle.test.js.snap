// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkSingle renders correctly with job 1`] = `
Array [
  <mock-Circles />,
  <div
    className="layout"
  >
    <div
      className="navWrapper"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/work"
        onClick={[Function]}
      >
        Work
      </a>
      <a
        href="/art"
        onClick={[Function]}
      >
        Art
      </a>
    </div>
    <div
      className="navClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="navOpen visible"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="hamburger.png"
      />
    </div>
    <a
      className="backLink"
      href="/work"
      onClick={[Function]}
    >
      <img
        alt="back"
        src="arrow.svg"
      />
    </a>
    <div
      className="section"
    >
      <div
        className="left"
      >
        <div
          className="body"
        >
          <div
            className="hideMobile"
          >
            <div
              className="topWrapper"
            >
              <div
                className="titleWrapper"
              >
                <div
                  className="title"
                >
                  test
                </div>
                <div
                  className="subtitle"
                >
                  test
                </div>
              </div>
              <a
                className="hideMobile"
                href="/test"
                onClick={[Function]}
              >
                <img
                  alt="external link to site"
                  className="link"
                  src="link.svg"
                />
              </a>
            </div>
          </div>
          <div
            className="midWrapper"
          >
            <div>
              - 
              lorem ipsum etc
            </div>
            <div>
              - 
              lorem ipsum esque
            </div>
          </div>
          <div
            className="bottomWrapper"
          >
            <div
              className="span"
            >
              test
            </div>
            <a
              className="showMobile"
              href="/test"
              onClick={[Function]}
            >
              <img
                alt="external link to site"
                className="link"
                src="link.svg"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="right"
      >
        <div
          className="showMobile"
        >
          <div
            className="topWrapper"
          >
            <div
              className="titleWrapper"
            >
              <div
                className="title"
              >
                test
              </div>
              <div
                className="subtitle"
              >
                test
              </div>
            </div>
            <a
              className="hideMobile"
              href="/test"
              onClick={[Function]}
            >
              <img
                alt="external link to site"
                className="link"
                src="link.svg"
              />
            </a>
          </div>
        </div>
        <div
          className="carousel-root"
        >
          <div
            className="carousel carousel-slider"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <ul
              className="control-dots"
            >
              <li
                aria-label="slide item 1"
                className="dot selected"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
                value={0}
              />
            </ul>
            <button
              aria-label="previous slide / item"
              className="control-arrow control-prev control-disabled"
              onClick={[Function]}
              type="button"
            />
            <div
              className="slider-wrapper axis-horizontal"
              style={Object {}}
            >
              <ul
                className="slider animated"
                style={
                  Object {
                    "MozTransitionDuration": "350ms",
                    "OTransform": "translate3d(-100%,0,0)",
                    "OTransitionDuration": "350ms",
                    "WebkitTransform": "translate3d(-100%,0,0)",
                    "WebkitTransitionDuration": "350ms",
                    "msTransform": "translate3d(-100%,0,0)",
                    "msTransitionDuration": "350ms",
                    "transform": "translate3d(-100%,0,0)",
                    "transitionDuration": "350ms",
                  }
                }
              >
                <li
                  className="slide selected previous"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="imageWrapper"
                  >
                    <img
                      alt="work 0"
                      className="carousel"
                      src="mock_long.jpg"
                      style={Object {}}
                    />
                  </div>
                </li>
                <li
                  className="slide selected previous"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="imageWrapper"
                  >
                    <img
                      alt="work 0"
                      className="carousel"
                      src="mock_long.jpg"
                      style={Object {}}
                    />
                  </div>
                </li>
              </ul>
            </div>
            <button
              aria-label="next slide / item"
              className="control-arrow control-next control-disabled"
              onClick={[Function]}
              type="button"
            />
            <p
              className="carousel-status"
            >
              1 of 1
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`WorkSingle renders correctly with project 1`] = `
Array [
  <mock-Circles />,
  <div
    className="layout"
  >
    <div
      className="navWrapper"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/work"
        onClick={[Function]}
      >
        Work
      </a>
      <a
        href="/art"
        onClick={[Function]}
      >
        Art
      </a>
    </div>
    <div
      className="navClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="navOpen visible"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="hamburger.png"
      />
    </div>
    <a
      className="backLink"
      href="/work"
      onClick={[Function]}
    >
      <img
        alt="back"
        src="arrow.svg"
      />
    </a>
    <div
      className="section"
    >
      <div
        className="left"
      >
        <div
          className="body"
        >
          <div
            className="hideMobile"
          >
            <div
              className="topWrapper"
            >
              <div
                className="titleWrapper"
              >
                <div
                  className="title"
                >
                  test
                </div>
                <div
                  className="subtitle"
                >
                  test
                </div>
              </div>
              <a
                className="hideMobile"
                href="/test"
                onClick={[Function]}
              >
                <img
                  alt="external link to site"
                  className="link"
                  src="link.svg"
                />
              </a>
            </div>
          </div>
          <div
            className="midWrapper"
          >
            <div>
              - 
              lorem ipsum etc
            </div>
            <div>
              - 
              lorem ipsum esque
            </div>
          </div>
          <div
            className="bottomWrapper"
          >
            <div
              className="span"
            >
              test
            </div>
            <a
              className="showMobile"
              href="/test"
              onClick={[Function]}
            >
              <img
                alt="external link to site"
                className="link"
                src="link.svg"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="right"
      >
        <div
          className="showMobile"
        >
          <div
            className="topWrapper"
          >
            <div
              className="titleWrapper"
            >
              <div
                className="title"
              >
                test
              </div>
              <div
                className="subtitle"
              >
                test
              </div>
            </div>
            <a
              className="hideMobile"
              href="/test"
              onClick={[Function]}
            >
              <img
                alt="external link to site"
                className="link"
                src="link.svg"
              />
            </a>
          </div>
        </div>
        <div
          className="carousel-root"
        >
          <div
            className="carousel carousel-slider"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <ul
              className="control-dots"
            >
              <li
                aria-label="slide item 1"
                className="dot selected"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
                value={0}
              />
            </ul>
            <button
              aria-label="previous slide / item"
              className="control-arrow control-prev control-disabled"
              onClick={[Function]}
              type="button"
            />
            <div
              className="slider-wrapper axis-horizontal"
              style={Object {}}
            >
              <ul
                className="slider animated"
                style={
                  Object {
                    "MozTransitionDuration": "350ms",
                    "OTransform": "translate3d(-100%,0,0)",
                    "OTransitionDuration": "350ms",
                    "WebkitTransform": "translate3d(-100%,0,0)",
                    "WebkitTransitionDuration": "350ms",
                    "msTransform": "translate3d(-100%,0,0)",
                    "msTransitionDuration": "350ms",
                    "transform": "translate3d(-100%,0,0)",
                    "transitionDuration": "350ms",
                  }
                }
              >
                <li
                  className="slide selected previous"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="imageWrapper"
                  >
                    <img
                      alt="work 0"
                      className="carousel"
                      src="mock_long.jpg"
                      style={Object {}}
                    />
                  </div>
                </li>
                <li
                  className="slide selected previous"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="imageWrapper"
                  >
                    <img
                      alt="work 0"
                      className="carousel"
                      src="mock_long.jpg"
                      style={Object {}}
                    />
                  </div>
                </li>
              </ul>
            </div>
            <button
              aria-label="next slide / item"
              className="control-arrow control-next control-disabled"
              onClick={[Function]}
              type="button"
            />
            <p
              className="carousel-status"
            >
              1 of 1
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
